src/utils/SafeTransferLib.sol

```diff
@@ -32,98 +34,108 @@ library SafeTransferLib {
         address to,
         uint256 amount
     ) internal {
-        bool success;
-
         assembly {
             // We'll write our calldata to this slot below, but restore it later.
             let memPointer := mload(0x40)
 
             // Write the abi-encoded calldata into memory, beginning with the function selector.
-            mstore(0, 0x23b872dd00000000000000000000000000000000000000000000000000000000)
-            mstore(4, from) // Append the "from" argument.
-            mstore(36, to) // Append the "to" argument.
-            mstore(68, amount) // Append the "amount" argument.
-
-            success := and(
-                // Set success to whether the call reverted, if not we check it either
-                // returned exactly 1 (can't just be non-zero data), or had no return data.
-                or(and(eq(mload(0), 1), gt(returndatasize(), 31)), iszero(returndatasize())),
-                // We use 100 because that's the total length of our calldata (4 + 32 * 3)
-                // Counterintuitively, this call() must be positioned after the or() in the
-                // surrounding and() because and() evaluates its arguments from right to left.
-                call(gas(), token, 0, 0, 100, 0, 32)
-            )
+            mstore(0x00, 0x23b872dd)
+            mstore(0x20, from) // Append the "from" argument.
+            mstore(0x40, to) // Append the "to" argument.
+            mstore(0x60, amount) // Append the "amount" argument.
+
+            if iszero(
+                and(
+                    // Set success to whether the call reverted, if not we check it either
+                    // returned exactly 1 (can't just be non-zero data), or had no return data.
+                    or(eq(mload(0x00), 1), iszero(returndatasize())),
```

Comment:
Before the call, we store the 4-byte selectors at `0x00`.
```solidity
// Equivalent to `mstore(0x00, 0x0000000000000000000000000000000000000000000000000000000023b872dd)`
mstore(0x00, 0x23b872dd)
```

Since all of the selectors (`0x23b872dd`, `0xa9059cbb`, `0x095ea7b3`) do not end with `0x01` (e.g. 0xffffff01`), the only way for the memory slot at `0x00` to be equal to `1` is for the call to write `1` to the first 32-bytes, which gives a `returndatasize()` that is at least 32.

Reply:
oh what the fuck this is genius